### PR TITLE
MM-52825 : Migrate "components/admin_console/database_settings.jsx" and tests to Typescript

### PR DESCRIPTION
**Pull request by mnj93**
#### Summary

Migrate "components/admin_console/database_settings.jsx" and tests to Typescript.  

#### Tasks Completed 

- [x] Rename the files to their corresponding TypeScript extensions (e.g., js to ts, jsx to tsx).
- [X] Update any components that utilize these files with the correct imports.
- [X] Migrate the associated tests, if applicable.
- [X] Address any errors raised by the TypeScript compiler. You can execute make check-types to view any encountered errors.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/23435
JIRA https://mattermost.atlassian.net/browse/MM-52825

#### Screenshots

None

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@mnj93 lets us know when this is ready for review

**Comment by mnj93** [comment]
@M-ZubairAhmed I am done with the changes but one check is failing, can you please help with this?

**Comment by M-ZubairAhmed** [comment]
@mnj93 you need to update the failing snapshots in jest

File: webapp/channels/src/components/admin_console/database_settings.tsx [hunk 1]
```diff
@@ -55,7 +76,7 @@ export default class DatabaseSettings extends AdminSettings {
         return res.ActiveSearchBackend;
     }
 
-    getStateFromConfig(config) {
+    getStateFromConfig(config: any) {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/database_settings.tsx:79]: can we get rid of any here

File: webapp/channels/src/components/admin_console/database_settings.tsx [hunk 2]
```diff
@@ -247,6 +270,7 @@ export default class DatabaseSettings extends AdminSettings {
                     onChange={this.handleChange}
                     setByEnv={this.isSetByEnv('SqlSettings.QueryTimeout')}
                     disabled={this.props.isDisabled}
+                    type='input'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/database_settings.tsx:273]: these probably should be text?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/database_settings.tsx (hunk 1 above)
Comment (mnj93) [reply webapp/channels/src/components/admin_console/database_settings.tsx:80]: fixed

Comment (mnj93) [reply webapp/channels/src/components/admin_console/database_settings.tsx]: fixed

File: webapp/channels/src/components/admin_console/database_settings.tsx [hunk 3]
```diff
@@ -29,16 +51,16 @@ export default class DatabaseSettings extends AdminSettings {
...
 
         config.SqlSettings.MaxIdleConns = this.parseIntNonZero(this.state.maxIdleConns);
         config.SqlSettings.MaxOpenConns = this.parseIntNonZero(this.state.maxOpenConns);
         config.SqlSettings.Trace = this.state.trace;
         config.SqlSettings.DisableDatabaseSearch = this.state.disableDatabaseSearch;
         config.SqlSettings.QueryTimeout = this.parseIntNonZero(this.state.queryTimeout);
-        config.SqlSettings.ConnMaxLifetimeMilliseconds = this.parseIntNonNegative(this.state.connMaxLifetimeMilliseconds);
-        config.SqlSettings.ConnMaxIdleTimeMilliseconds = this.parseIntNonNegative(this.state.connMaxIdleTimeMilliseconds);
+        config.SqlSettings.ConnMaxLifetimeMilliseconds = this.parseIntNonZero(this.state.connMaxLifetimeMilliseconds);
+        config.SqlSettings.ConnMaxIdleTimeMilliseconds = this.parseIntNonZero(this.state.connMaxIdleTimeMilliseconds);
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/database_settings.tsx:63]: I dont think we need to change the method here, lets keep the previous method but we can change the parseIntNonNegative param types

File: webapp/channels/src/components/admin_console/admin_settings.tsx [hunk 4]
```diff
@@ -160,7 +160,7 @@ export default abstract class AdminSettings <Props extends BaseProps, State exte
         return n;
     };
 
-    private parseIntNonNegative = (str: string, defaultValue?: number) => {
+    public parseIntNonNegative = (str: string, defaultValue?: number) => {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/admin_settings.tsx:163]: As mentioned in this 

We can allow the str param to be either a string or number and subsequently, in the function body check if its a string, parse it to number or if its a number take it as it is

File: webapp/channels/src/components/admin_console/database_settings.tsx [hunk 5]
```diff
@@ -10,17 +10,39 @@ import {t} from 'utils/i18n';
...
-export default class DatabaseSettings extends AdminSettings {
-    constructor(props) {
+interface Props {
+    license: {
+        IsLicensed: string;
+    };
+    isDisabled: boolean;
+}
+
+interface ComponentState extends BaseState {
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/database_settings.tsx:30]: nit
```suggestion
interface State extends BaseState {
```

File: webapp/channels/src/components/admin_console/database_settings.tsx [hunk 6]
```diff
@@ -10,17 +10,39 @@ import {t} from 'utils/i18n';
...
-import AdminSettings from './admin_settings';
+import AdminSettings, {BaseState} from './admin_settings';
 import BooleanSetting from './boolean_setting';
 import RequestButton from './request_button/request_button';
 import SettingsGroup from './settings_group';
 import TextSetting from './text_setting';
 
 import MigrationsTable from './database';
 import {DocLinks} from 'utils/constants';
+import {AdminConfig} from '@mattermost/types/config';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/database_settings.tsx:21]: nit: import placement

Place it after 
```
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {AdminConfig} from '@mattermost/types/config';

import {recycleDatabaseConnection, ping} from 'actions/admin_actions';
.
.
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by mnj93** [comment]
@M-ZubairAhmed snapshot test still failing for this component, I tried running test suit locally but was not able to, can you help me how this snapshot issue can be resolved?

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed snapshot test still failing for this component, I tried running test suit locally but was not able to, can you help me with how this snapshot issue can be resolved?

Oh they don't look to fail but yeah here is the steps
```sh
cd channels
npm run test -- -u
```

**Comment by mnj93** [comment]
[This](https://github.com/mattermost/mattermost/actions/runs/5648470660/job/15311521711?pr=23939) check is failing. 

Also tried running `npm run test -- -u` but its failing with this error for all the test cases

Screenshot 2023-07-25 at 12 01 43 PM

**Comment by mnj93** [comment]
@M-ZubairAhmed Hey, anything pending from my end for this PR?

**Comment by M-ZubairAhmed** [comment]
@mnj93 can you please update the branch

**Comment by M-ZubairAhmed** [comment]
@yasserfaraazkhan can you recheck if the e2e for this passes?

**Comment by M-ZubairAhmed** [comment]
@mnj93 can you check the failing CI

**Comment by mnj93** [comment]
> [This](https://github.com/mattermost/mattermost/actions/runs/5648470660/job/15311521711?pr=23939) check is failing.
> 
> Also tried running `npm run test -- -u` but its failing with this error for all the test cases
> 
> Screenshot 2023-07-25 at 12 01 43 PM

@M-ZubairAhmed Still facing this error when I try to run the tests.

**Comment by M-ZubairAhmed** [comment]
> > [This](https://github.com/mattermost/mattermost/actions/runs/5648470660/job/15311521711?pr=23939) check is failing.
> > Also tried running `npm run test -- -u` but its failing with this error for all the test cases
> > Screenshot 2023-07-25 at 12 01 43 PM
> 
> @M-ZubairAhmed Still facing this error when I try to run the tests.

This error is not produced in the CI, you can verify. Its just the snapshot update error. There is something not correct with your local setup for it to have that error.

**Comment by mnj93** [comment]
@M-ZubairAhmed Yeah and for updating snapshot I guess I need to run this command but its not working on my local setup.  Is there any other way to update the snapshot?

**Comment by hanzei** [comment]
@mnj93 Heads up that I updated the snapshots for you.

@yasserfaraazkhan The PR is ready for your review.

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by hanzei** [comment]
Thanks for the contribution @mnj93 :tada:

**Pull request merged**